Sources/Opus/Opus.Decoder.swift

```diff
@@ -73,30 +109,58 @@ public extension Opus.Decoder {
 // MARK: Private decode methods
 
 extension Opus.Decoder {
-	private func decode(_ input: UnsafeBufferPointer<UInt8>, to output: UnsafeMutableBufferPointer<Int16>) throws -> Int {
-		let decodedCount = opus_decode(
-			decoder,
-			input.baseAddress!,
-			Int32(input.count),
-			output.baseAddress!,
-			Int32(output.count),
-			0
-		)
+	private func decode(_ input: UnsafeBufferPointer<UInt8>,
+	                    to output: UnsafeMutableBufferPointer<Int16>,
+	                    packetSize: Int32?) throws -> Int
+	{
+		var decodedCount: Int32 = 0
+		if let size = packetSize {
+			decodedCount = opus_custom_decode(
+				decoder,
+				input.isEmpty ? nil : input.baseAddress,
+				size,
+				output.baseAddress!,
+				Int32(output.count)
+			)
+		} else {
+			decodedCount = opus_decode(
+				decoder,
+				input.isEmpty ? nil : input.baseAddress,
+				Int32(input.count),
+				output.baseAddress!,
+				Int32(output.count),
+				0
+			)
+		}
 		if decodedCount < 0 {
 			throw Opus.Error(decodedCount)
 		}
 		return Int(decodedCount)
 	}
 
-	private func decode(_ input: UnsafeBufferPointer<UInt8>, to output: UnsafeMutableBufferPointer<Float32>) throws -> Int {
-		let decodedCount = opus_decode_float(
-			decoder,
-			input.baseAddress!,
-			Int32(input.count),
-			output.baseAddress!,
-			Int32(output.count),
-			0
-		)
+	private func decode(_ input: UnsafeBufferPointer<UInt8>,
+	                    to output: UnsafeMutableBufferPointer<Float32>,
+	                    packetSize: Int32?) throws -> Int
+	{
+		var decodedCount: Int32 = 0
+		if let size = packetSize {
```

Comment:
If possible, this should branch on the presence of a non-nil `customMode` instance variable, which implies the decoder was initialized with a custom frame size.

Reply:
I've implemented this a different way; using the optional presence of a custom frame size.

Reply:
When decoding a stream of packets, is the frame size consistent, or does it vary? If it’s consistent, then this parameter shouldn’t exist.

Reply:
The size can be multiples of the underlying frame size

Reply:
Input.count can be zero, which means a dropped packet. The decoder still needs to know how much data was dropped, so we assume a single packet. 
Perhaps the case of multiples of underlying frame size is something that needs to be reconsidered. I'll need to investigate a bit first though.

Reply:
OK, great. Then can that be taken from the `customFrameSize` instance variable?